src/gettext.cpp

```diff
@@ -164,12 +164,52 @@ static void MSVC_LocaleWorkaround(int argc, char* argv[])
 
 #endif
 
+namespace {
+	std::string system_language;
+}
+
 /******************************************************************************/
 void init_gettext(const char *path, const std::string &configured_language,
 	int argc, char *argv[])
 {
 #if USE_GETTEXT
-	// First, try to set user override environment
+	char *system_language_c = getenv("LANGUAGE");
+	if (system_language_c == NULL)
```

Comment:
This should also check for the empty string and the value of `LC_ALL`.

Reply:
I can't get LC_ALL to work in the master branch, so it appears that it's ignored when it comes to detecting the system language

```
unset LANGUAGE
unset LANG
LC_ALL=de_DE.UTF-8 ./minetest  --verbose
```

Reply:
I seem to get very interesting results with this.

My current setup has `en_US` and the `de_DE` locales (and some more), but not `fr_FR` listed in `locale -a`, and:

```sh
# LANGUAGE= (empty string), LANG=en_US.UTF-8
env LC_ALL=de_DE.UTF-8 bin/minetest # UI appears in German, same for LANG=... and LANGUAGE=...
env LC_ALL=fr_FR.UTF-8 bin/minetest # UI appears in English
env LANG=fr_FR.UTF-8 bin/minetest # UI appears in English
env LANGUAGE=fr_FR.UTF-8 bin/minetest # UI appears in French

unset LANGUAGE
# same as above

unset LANG
env LC_ALL=de_DE.UTF-8 bin/minetest # UI appears in German, same for LANG=...
env LANGUAGE=de_DE.UTF-8 bin/minetest # UI appears in English (!)
env LC_ALL=fr_FR.UTF-8 bin/minetest # UI appears in English
env LANG=fr_FR.UTF-8 bin/minetest # UI appears in English
env LANGUAGE=fr_FR.UTF-8 bin/minetest # UI appears in English

export LANGUAGE=
# same as above
```

So (at least on my computer) it seems like setting `LC_ALL` or `LANG` to a language not listed in `locale -a` does not work, while setting `LANGUAGE` only works when `LANG` is set to a known (?) locale.

It should also be noted that `env LC_ALL=fr_FR.UTF-8 locale` (and `env LANG=fr_FR.UTF-8 locale`) throws the errors
```
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
```
while `env LANGUAGE=fr_FR.UTF-8 locale` does not.

More interestingly, the result is the same when using `gettext --version` instead of `bin/minetest`, so 🫠

That said, I would assume this is somewhat related to the first entry in https://www.gnu.org/software/gettext/FAQ.html#integrating_noop.

IMO the sane behavior here would be to set the language based on `LANGUAGE`, `LC_ALL`,  `LANG` (in that order) and let gettext figure out the rest.

---

Some more information on my setup:
-  Arch Linux
- `gettext --version`: `gettext (GNU gettext-runtime) 0.22.4`
- `locale --version`: `locale (GNU libc) 2.38`

Reply:
> So (at least on my computer) it seems like setting LC_ALL or LANG to a language not listed in locale -a does not work, while setting LANGUAGE only works when LANG is set to a known (?) locale.

Ahhh that'll be why, my `locale -a` only has C, en, and POSIX. I'll see how to install a language so I can test this. Thanks for looking into this

Reply:
Given that we only ever set LANGUAGE, it might be enough to just reset that env var and rely on gettext's logic on reading other env vars. I'll work out how to test LC_ALL and see if this approach works

Reply:
we might have to unset the other variables however depending on priority order

Reply:
It seems like (re)setting `LANGUAGE` alone is enough, and gettext does read other environment variables when `LANGUAGE` is set to the empty string. (At least it works for me)

That said, I do not have a Windows machine to test the MSVC workaround, so I don't know what happens when `LANGUAGE` is reset to the empty string there.

> we might have to unset the other variables however depending on priority order

`LANGUAGE` is [at the top of the priority order](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html) so that is not a problem.